WebRoot/js/zimbraMail/share/model/ZmList.js

```diff
@@ -730,6 +737,49 @@ function(params) {
 	}
 };
 
+/**
+ * Check if deleting item(s) is opened
+ *
+ * @param {Array}   toDelete   list of items to delete
+ */
+ZmList.prototype.isDeletingItemOpen =
+function(toDelete) {
+	var deletingItemIds = [];
+	for (var i = 0; i < toDelete.length; i++) {
+		deletingItemIds.push(toDelete[i].id);
+		if (toDelete[i].msgIds) {
+			for (var j = 0; j < toDelete[i].msgIds.length; j++) {
+				deletingItemIds.push(toDelete[i].msgIds[j]);
+			}
+		}
+	}
+
+	var viewIdToIgnore;
+	if (
+		appCtxt.getCurrentView() instanceof ZmMailMsgView ||
+		appCtxt.getCurrentView() instanceof ZmConvView2 ||
+		appCtxt.getCurrentView() instanceof ZmEditContactView
+	) {
+		viewIdToIgnore = appCtxt.getCurrentView().getController().tabId;
+	}
+	var appViewMgr = appCtxt.getAppViewMgr();
+	return appViewMgr && appViewMgr.isDeletingItemOpen(deletingItemIds, null, viewIdToIgnore);
+};
+
+/**
+ * Show a warning message dialog for item hard deletion
+ */
+ZmList.prototype._popupBlockItemDeletionWarningDialog =
```

Comment:
Not sure. 
I think this is the general dialog so Can we move it to the `ZmAppViewMgr.js` ?

Reply:
Moved the function to ZmAppViewMgr.